src/liballoc/vec.rs

```diff
@@ -962,13 +963,15 @@ impl<T> Vec<T> {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn swap_remove(&mut self, index: usize) -> T {
+        let len = self.len();
+        assert!(index < len, "index out of bounds: the len is {} but the index is {}", len, index);
```

Comment:
Using formatting in the error [seems to cause performance issues](https://github.com/rust-lang/rust/pull/70573#issuecomment-607364101); let me know if you'd prefer an unformatted message.